public/app/features/dashboard-scene/scene/LibraryPanelBehavior.tsx

```diff
@@ -61,8 +55,20 @@ export class LibraryPanelBehavior extends SceneObjectBase<LibraryPanelBehaviorSt
 
     this.setState({ _loadedPanel: libPanel, isLoaded: true, name: libPanel.name, title: libPanelModel.title });
 
-    const clone = vizPanel.clone(vizPanelState);
-    gridItem.setState({ body: clone });
+    vizPanel.setState(vizPanelState);
+    vizPanel.changePluginType(libPanelModel.type, vizPanelState.options, vizPanelState.fieldConfig);
```

Comment:
The changes look good to me! I'm testing hard on this one. This function was causing some bugs in the past, where some properties were lost when changing between types. Since here we're changing the type to library panel, I wonder if we break something. @mdvictor actually helped me with that one

Reply:
@ivanortegaalba yea, seems to work. if we run into problems we could try to init the VizPanel into an empty state with a dummy panel (in save model to scene) so that we always change plugin here.

Reply:
Indeed, this is great!!! Works good and removes that clone which is fantastic!